docs/home-automation/home-assistant/supervisor-addons/android-debug-bridge/adb.md

```diff
@@ -0,0 +1,336 @@
+
+```YAML
+media_player:
+
+  - platform: androidtv
+    host: 1.2.3.4
+    name: "Living Room Super Shield"
+    adb_server_ip: 127.0.0.1
+    adb_server_port: 5037
+```
+4. Restart Home Assistant
+5. Restart `ADB` in Home Assistant
+6. Allow `Home Assistant` to access to your device, remember the check the box to always remember this device
+
+
+## Connect to the container running ADB
+
+If you are using Home Assistant backed Ubuntu or other debian based OS this next step is done in Bash, if you are running the enclosed Home Assistant OS install the [Terminal & SSH plugin](https://github.com/home-assistant/hassio-addons/tree/master/ssh)
+
+Open you command line `Shell` to bash into the container
+
+    docker exec -it addon_a0d7b954_adb /bin/bash
+
+### List all installed Packages and Activities
+
+This section explains how you can query `ADB` for applications running on you `Android TV`
+
+1. Connect to ADB Shell 
+
+        adb -s 1.2.3.4:5555 shell
+
+2. Type in the following command to list all packages and their associated files. 
+
+        pm list packages
+
+3. To filter the output based on the package name
+
+        pm list packages | grep com.amazon.amazonvideo.livingroom.nvidia
+
+4. To find all activities published by a package, use the following command & replace com.symbol.wfc.voice by the name of the package to process.
+
+        dumpsys package | grep -Eo "^[[:space:]]+[0-9a-f]+[[:space:]]+com.nvidia.bbciplayer.launch/[^[:space:]]+" | grep -oE "[^[:space:]]+$"
+
+
+### List installed Packages and Activities
+
+        pm list packages | sed -e "s/package://" | while read x; do cmd package resolve-activity --brief $x | tail -n 1 | grep -v "No activity found"; done 
+
+</br> Output
+
+    com.plexapp.android/com.plexapp.plex.activities.SplashActivity
+    android/com.android.internal.app.ResolverActivity
+    com.nvidia.tegrazone3/com.nvidia.tegrazone.MainActivity
+    com.android.documentsui/.LauncherActivity
+    com.nest.android/com.obsidian.v4.activity.LoginActivity
+    com.silicondust.view/.App
+    com.plexapp.mediaserver.smb/com.plexapp.mediaserver.ui.main.MainActivity
+    com.android.vending/com.google.android.finsky.tvmainactivity.TvMainActivity
+    nextapp.fx/.ui.ExplorerActivity
+    com.onemainstream.skynews.android/.common.splash.SplashActivity
+    com.google.android.tv/com.android.tv.TvActivity
+    com.nvidia.inputviewer/.AccessorySelectionActivity
+    com.android.gallery3d/.app.GalleryActivity
+    com.nvidia.ota/.ui.LauncherActivity
+    com.ubnt.unifi.protect/com.ubnt.sections.splash.SplashActivity
+    com.google.android.youtube.tv/com.google.android.apps.youtube.tv.activity.ShellActivity
+    com.google.android.tv.remote.service/.settings.SettingsActivity
+    com.valvesoftware.steamlink/.SteamShellActivity
+    com.netflix.ninja/.MainActivity
+
+
+Make a list of your applications and and the options available, you'll need this later. I have exported a few Apps below for the UK users :)
```

Comment:
I can't see much here which is specific to UK, besides a couple of the services? 

```suggestion
Make a list of your applications and and the options available, you'll need this later.
```

Reply:
Rephrased this line